lib/lz4.c

```diff
@@ -1708,10 +1719,9 @@ LZ4_decompress_generic(
             /* get matchlength */
             length = token & ML_MASK;
 
-            if ((checkOffset) && (unlikely(match + dictSize < lowPrefix))) { goto _output_error; } /* Error : offset outside buffers */
```

Comment:
So, if I understand correctly,
this check has been moved forward into the complex, and replaced by 2 instances of the same test.

While I guess it works correctly, I wonder what's the need for it ?

Reply:
Hi @Cyan4973 :
If you profiling this condition check invoked number comparison between 1.9.x and 1.8.3.
You will see the condition check difference via lzbench with dickens test case.
v1.8.3 34959
v.1.9.x 1055885

After investigate the code, we could see the difference.
v.1.8.3 SKIP the condition check if 
if condition is true in:
https://github.com/lz4/lz4/blob/v1.8.3/lib/lz4.c#L1463
AND below condition is true
https://github.com/lz4/lz4/blob/v1.8.3/lib/lz4.c#L1478
The **"if ((checkOffset) && (unlikely(match + dictSize < lowPrefix))) goto _output_error;   /* Error : offset outside buffers */"** should not be invoked.

v1.9.3
The "if ((checkOffset) && (unlikely(match + dictSize < lowPrefix))) goto _output_error;   /* Error : offset outside buffers */" code will be invoked in every loop which lead to downgrade.
So the fix would be move this check to specific condition to avoid useless condition check.

After this change, the call number is same as v1.8.3
Chenxi